Source/DafnyCore/AST/Types.cs

```diff
@@ -2475,6 +2485,29 @@ public UserDefinedType(TypeParameter tp)
     }
   }
 
+  public override bool PartiallySupportsEquality {
+    get {
+      var totalEqualitySupport = SupportsEquality;
+      if (!totalEqualitySupport && ResolvedClass is IndDatatypeDecl dt && dt.IsRevealedInScope(Type.GetScope())) {
+        // Equality is partially supported for:
+        // an inductive datatype with at least one non-ghost constructor where every argument of a non-ghost constructor
+        // totally supports equality.
+        var hasNonGhostConstructor = false;
+        foreach (var ctor in dt.Ctors) {
+          if (!ctor.IsGhost) {
+            hasNonGhostConstructor = true;
+            if (!ctor.Formals.All(formal => !formal.IsGhost && formal.Type.SupportsEquality)) {
```

Comment:
No, because if an argument is ghost, it is not available at run time, so then it can't be compared. (Note that all occurrences of "equality support" and "partial equality support" refers to _compiled_ support of equality, that is, run time support.)

I expanded the comment in the code that try to better clarify these things.